### ISSUE TITLE
Enhancement: improve markdown links to FAQ entries

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/temp-test-ioccc#2417]
### Is there an existing issue for this?

- [X] I have searched for existing issues and did not find anything like it

### Describe the problem

In regards to markdown links to FAQ entries:

- Some "_link text_"s are too verbose.
- Some "_link text_"s do not contain enough text identify the topic.
- Some "_link text_"s  refer to the FAQ x.y number.
- Some "_link text_"s  include a trailing _?_ (question mark).
- Some "_link text_"s are surrounded by _"_ s (quotes).
- Some "_link text_"s  do not fit into the flow of text surrounding them.
- Some "_link URLs_"s, outside of the `faq.md` file, use a name tag of the form "_#faqX_" or "_#faqX_Y_".

### Relevant images, screenshots or other files

Consider these FAQ entry markdown links:

```
contact.md: * **Fix a winning IOCCC entry** - See [FAQ 5.2](faq.html#fix_an_entry)

index.md: See [FAQ 0.0: How may I submit to the IOCCC?](faq.html#submit) for more information.

thanks-for-help.md: See [FAQ 4.6  - Why was arg count and/or type changed in main&#x28;&#x29; in some older entries?](faq.html#faq4_6) for more details.

status.md: are in the process of judging the [submissions](faq.html#faq2_0) they received while

news.md: See "[FAQ 6.12: What is Mastodon and why does IOCCC use it?](faq.html#try_mastodon)" for more
```

### What you expect

Regarding markdown links to FAQ entries:

- ALL URLs of markdown links, outside of the `faq.md` file use "_faq.html#textstring_".
- The URLs of markdown links in of the table of contents in the `faq.md` file, use  "_#faqX_" or "_#faqX_Y_".
- The URLs of markdown links outside of the table of contents in the `faq.md` file, use  "_#textstring_".
- No "_link text_"s mention the FAQ X.Y entry number.
- The "_link text_"s are just one or a few words in length.
- The few words of the "_link text_" fit into the flow of words or sentence of context the are used within.

### Environment

```markdown
n/a
```

### Relevant links

n/a

### Anything else?

See also [comment-2161538434](https://github.com/ioccc-src/temp-test-ioccc/issues/2006#issuecomment-2161538434), and [comment-2161543051](https://github.com/ioccc-src/temp-test-ioccc/issues/2006#issuecomment-2161543051), and [comment-216169570](https://github.com/ioccc-src/temp-test-ioccc/issues/2006#issuecomment-2161695708).

## What this issue NOT is about

- Adding new FAQs is NOT in scope.

- Improving existing FAQs is NOT in scope.

- Improving the organization of the FAQ set NOT in scope.

By the above we don't mean those things shouldn't be done. In some cases they do need to be done.  We just mean this isn't the issue to do that.

We do plan to create a new **improve the organization of the FAQ** once this issue is resolved, for example.

**Comment by xexyl** [comment]
Oh I like how the templates I added turned out!

Please assign this to me too and I'll look at it later - as you say, after #2006 is done.

I might add that unrelated I **STILL** have a problem with the IP changes. My registrar has problems with glue record set up and if you have IPv6 addresses you have to contact them. Well the old addresses, IPv6 and IPv4, for the slave DNS server that still exists (I was able to delete the ones that no longer exist) have the old IPs so I just sent them an email ... I decided to not bother with the new IPv6 address as that would mean I would have to email them again.

This way it is hoped that when I get the new IPs (hopefully in the not too distant future) I can fix it myself. I am also thinking about seeing if I can get DNSSEC working again. I must also now fix another problem (simple one at least that will take very little time: just updating one of the NS RRs in some of the zone files). 

I am not sure what I can do here at the repo today but I hope to at least resume where I left off yesterday.

Anyway please do assign this task to me when you see this message: for I know you can't assign an issue to someone until they reply.

**Comment by xexyl** [comment]
One thing I noticed here though is it looks like the tickbox of 'I have searched for ..' makes it like a task list. Hmm ... or did I miss a list you added? I can believe that as I didn't look at it in full and I'm not completely awake either but if it did do that we might want to look at doing it differently, maybe? What do you think? It could just be a text box that they say 'yes' to or something so as to not confuse it as a task list?

**Comment by lcn2** [comment]
> One thing I noticed here though is it looks like the tickbox of 'I have searched for ..' makes it like a task list. Hmm ... or did I miss a list you added? I can believe that as I didn't look at it in full and I'm not completely awake either but if it did do that we might want to look at doing it differently, maybe? What do you think? It could just be a text box that they say 'yes' to or something so as to not confuse it as a task list?

That it is a checkbox ✅ isn't a problem. And it looks nice, don't you think?

Yes, the template looks good 👍

**Comment by xexyl** [comment]
> > One thing I noticed here though is it looks like the tickbox of 'I have searched for ..' makes it like a task list. Hmm ... or did I miss a list you added? I can believe that as I didn't look at it in full and I'm not completely awake either but if it did do that we might want to look at doing it differently, maybe? What do you think? It could just be a text box that they say 'yes' to or something so as to not confuse it as a task list?
> 
> That it is a checkbox ✅ isn't a problem. And it looks nice, don't you think?
> 
> Yes, the template looks good 👍 

I think it looks nice but I also think it's unfortunate that it thinks that it's a task list. 

But I agree that the tickbox is a good idea too. I just wish it didn't interpret it as a task list. That might be misleading. 

Okay okay! The irony is of course this is for the IOCCC 😀 but still not everything needs to be obscure .. :-) 

Thanks for assigning. I will look at it after the top priority issue is done. If however any part of that leads me to making slight fixes here then I will but I won't actually work on fixing the whole thing until after the other issue. 

Though of course maybe when you are done with your other things that you have been forced to do you might also look into this. 

Too bad this can't all be automated. Some of it could be but it's too convoluted to do it completely. At least once the tags are in the links can be updated easily. But of course the name for the links should be consistent and that can't be automated in every case for reasons I brought up.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by lcn2** [comment]
Our suggestion is that the FAQ not be reorganized at this time: certainly not until this issue is resolved. 

Later, if there is time after this issue is resolved and before code freeze (say because the screenshots for the submit server and registration are not ready) a FAQ reorganization can be done.

**Comment by xexyl** [comment]
> Our suggestion is that the FAQ not be reorganized at this time: certainly not until this issue is resolved.
> 
> Later, if there is time after this issue is resolved and before code freeze (say because the screenshots for the submit server and registration are not ready) a FAQ reorganization can be done.

Sure. It can always come later .. or not.

**Comment by lcn2** [comment]
## We see the value in reorganizing the FAQ

Let's decide on how FAQs should be externally referenced, given the [constraints in the top comment](https://github.com/ioccc-src/temp-test-ioccc/issues/2417) soon.

Once that decision is made, then we can change all references to FAQs.

Once all the references to FAQs have changed, reorganizing the FAQ is an easy process of reorganizing the FAQ table of contents and moving sections around in the FAQ.

**Comment by xexyl** [comment]
> ## We see the value in reorganizing the FAQ
> Let's decide on how FAQs should be externally referenced, given the [constraints in the top comment](https://github.com/ioccc-src/temp-test-ioccc/issues/2417) soon.
> 
> Once that decision is made, then we can change all references to FAQs.
> 
> Once all the references to FAQs have changed, reorganizing the FAQ is an easy process of reorganizing the FAQ table of contents and moving sections around in the FAQ.

Glad you have decided reorganising it is a good idea, after all! I think it'll help. I cannot look at that comments right now but maybe in a bit. I was actually hoping I could have done something by now but not having much luck yet. I will be going afk for a little bit soon and maybe when back I can focus more. At least it's still early.

One question I can pose now, however, is this. I think I brought it up before but this should come after #2006 but before the merge, right? In other words: I should still focus on the other one but if by chance I have the time or thought on some of this I can work on this too. Of course discussing it is also a good idea and obviously necessary first but the point is this is less important than the other one so it's okay if I kind of let this sit a bit (discuss but not do), right?

I'll be going afk now but I hope that when I'm back I can start looking at things better.

**Comment by xexyl** [comment]
Hmm ...

```
< 8. NOTE: item 8 was removed by the judges so as to not spoil the challenge of
< figuring how arguments can be used to alter the exit value.
```

shouldn't that be there actually?

**Comment by xexyl** [comment]
> Hmm ...
> 
> ```
> < 8. NOTE: item 8 was removed by the judges so as to not spoil the challenge of
> < figuring how arguments can be used to alter the exit value.
> ```
> 
> shouldn't that be there actually?

In the archive it's 7 and it was also removed .. so no way to restore it seemingly? That's unfortunate.

**Comment by xexyl** [comment]
Oh .. wrong place. Oops!

**Comment by lcn2** [comment]
> One question I can pose now, however, is this. I think I brought it up before but this should come after #2006 but before the merge, right? 

Yes. 

And we might find a reason to add more FAQ.  Better to reorganize after adding stuff. 

> In other words: I should still focus on the other one but if by chance I have the time or thought on some of this I can work on this too. Of course discussing it is also a good idea and obviously necessary first but the point is this is less important than the other one so it's okay if I kind of let this sit a bit (discuss but not do), right?

Sure.  And a discussion here about how to reference the FAQs ahead of time (and before finishing issue #2006) might be useful too.

**Comment by xexyl** [comment]
> > One question I can pose now, however, is this. I think I brought it up before but this should come after #2006 but before the merge, right?
> 
> Yes.
> 
> And we might find a reason to add more FAQ. Better to reorganize after adding stuff.
> 

Yes. Like the things I suggested yesterday .. or whenever it was. Monday if not yesterday.

> > In other words: I should still focus on the other one but if by chance I have the time or thought on some of this I can work on this too. Of course discussing it is also a good idea and obviously necessary first but the point is this is less important than the other one so it's okay if I kind of let this sit a bit (discuss but not do), right?
> 
> Sure. And a discussion here about how to reference the FAQs ahead of time (and before finishing issue #2006) might be useful too.

Well actually I was holding off on looking at the FAQ until everything else in #2006 is done for the very reason that things might be added or changed (and there have been numerous additions and changes) but yes it might be a good idea to talk about this ahead of that.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by xexyl** [comment]
Please feel free to remind me of this soon if I don't get to it. Have to go for the day.

**Comment by xexyl** [comment]
As far as some of the issues noted I either don't see it as a problem or I misunderstand. For instance:

```
Some "link text"s include a trailing ? (question mark).
```

What is the problem with a question mark when it's a question? Or is it more than one?

By:

```
Some "link text"s refer to the FAQ x.y number.
```

.. you mean not in the table of contents but down in the actual questions? What should it look like - for each part, the list and the questions themselves?

In fact what are `link texts`? Do you mean when linking TO it or the actual link that you are taken to when going to that question?

Maybe there are more that I'm not clear on but instead I'll say I'm not sure I see the problems in the relevant screenshots (or 'screenshots') section. Ah .. maybe that's the idea I had (I think I had it) where it should say something short that links TO the FAQ but not the FAQ file itself? Given that they're not in the faq.md itself I presume that's it.

If that's the case perhaps my questions above are not in the right context. The only one that would seem to not be a problem, maybe, is the question mark - it should I think?

That's all I can say for the time being but hopefully it starts some discussion. Hopefully tomorrow will be a day where I will be clearer. Will be having food soon probably and mostly done for the day.

**Comment by xexyl** [comment]
Okay I read again:

> In regards to markdown links to FAQ entries:

... so definitely linking TO the FAQ and not in the FAQ itself. That's reason enough for me to stop for the day. I guess only one question I have above - the question mark one - is still relevant. Back tomorrow hopefully to work on the other issue but maybe I can fit some thoughts in this one too.

**Comment by lcn2** [comment]
> What is the problem with a question mark when it's a question? Or is it more than one?

If one wishes to reference a FAQ within some sentence, one may not want the FAQ link to contain a question mark.  Punctuation is the preview of the sentence, not some FAQ link term that might appear within the sentence.

Say you wish to write the sentence:

```
The `.entry.json` file also contains refers to the "author handle" of the authors who wrote the entry.
```

Where the "author handle" is a link to the "FAQ 6.5: What is an author handle?" FAQ.  It would create awkward punctuation if one simply inserted that FAQ string into the middle of that sentence:

```
The `.entry.json` file also contains refers to the ["FAQ 6.5: What is an author handle?"](faq.html#author_handle) of the authors who wrote the entry.
```

The FAQ punctuation (the quote marks and the final question mark) gets in the way of the sentence one is trying to construct.  It is better to write:

```
The `.entry.json` file also contains refers to the [author_handle](faq.html#author_handle) of the authors who wrote the entry.
```

Let the FAQ link contain a sword or very short phrase.  Let the sentence that needs to link to some FAQ control and select the punctuation it needs.

**Comment by xexyl** [comment]
> > What is the problem with a question mark when it's a question? Or is it more than one?
> 
> If one wishes to reference a FAQ within some sentence, one may not want the FAQ link to contain a question mark. Punctuation is the preview of the sentence, not some FAQ link term that might appear within the sentence.
> 
> Say you wish to write the sentence:
> 
> ```
> The `.entry.json` file also contains refers to the "author handle" of the authors who wrote the entry.
> ```
> 
> Where the "author handle" is a link to the "FAQ 6.5: What is an author handle?" FAQ. It would create awkward punctuation if one simply inserted that FAQ string into the middle of that sentence:
> 
> ```
> The `.entry.json` file also contains refers to the ["FAQ 6.5: What is an author handle?"](faq.html#author_handle) of the authors who wrote the entry.
> ```
> 
> The FAQ punctuation (the quote marks and the final question mark) gets in the way of the sentence one is trying to construct. It is better to write:
> 
> ```
> The `.entry.json` file also contains refers to the [author_handle](faq.html#author_handle) of the authors who wrote the entry.
> ```
> 
> Let the FAQ link contain a sword or very short phrase. Let the sentence that needs to link to some FAQ control and select the punctuation it needs.

I agree with the quotes problem .. did not even know that was there. But the question mark - I think that depends on the item in question. I'm not sure what you mean by the examples though so it's harder to understand. Would you clarify what you mean by:

```
contains refers
```

(obviously a typo but I want to be sure before I try and parse what you're getting at.)

Thanks.

**Comment by xexyl** [comment]
> > In fact what are link texts? Do you mean when linking TO it or the actual link that you are taken to when going to that question?
> 
> The link text is the string that is the link. For example:
> 
> ```
> This sentence has a [link text](http://example.com) in the middle of it.
> ```

That is what I wondered and thought but wanted confirmation. Thanks. That does make sense.

> 
> The "link text" is the link text.

Thanks.
> 
> One does **NOT** want to put "FAQ 6.7" in some link text such as:
> 
> ```
> See the [FAQ 6.7](faq.html#entry_id) for more information about the syntax of an entry's identifying string. 
> ```

I agree.

> 
> If one needed to renumber that FAQ from 6.7 to, say 4.2 because one wanted to re-organize the FAQ, the presence of that "FAQ 6.7" would require that sentence of to modified when thr FAQ was re-organized.
> 

I agree.

> Instead if one did:
> 
> ```
> See the "[entry id](faq.html#entry_id) FAQ" for more information about the syntax of an entry's identifying string.
> ```
> 
> There, without specifying the "6.7" FAQ number: reorganizing the FAQ would **NOT** require that sentence to be changed because heighten the "6.7" FAQ number is neither in the "link text", nor in the link URL.
> 

Yes .. I suggested that a while back but of course I've only done a few as I've been working on the other html files that are higher priority.

> And regarding the previous comment about not adding punctuation (neither the quotes nor the question mark): notice how the FAQ link text does **NOT** contain any punctuation. The sentence that contains the link to the FAQ is in full punctuation control.


Okay well yes with this change in place that indeed won't matter and the above comment does not need any clarification (with the typo).

I think this will be much nicer. 

Am I right though in saying I should worry about the FAQ later on except that if I come across something that is relevant to the other issues I should (of course) update it?

What I have to do I noted in the issue about less obfuscated versions and one thing I did not note but need to do is to check the thanks file for those entries as they had to be changed some too. But since the idea of the directory name and maybe the Makefile has to be considered it might be better to wait before I do that anyway. 

Off to do other things now. Hopefully tomorrow there will be some replies to some of the pending issues so more work can be done. Just going to take it easy the rest of the day I think.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by xexyl** [comment]
# Question about this wrt var.mk

It has this:


```makefile

##########################
# SD1 and SDL2 libraries #
##########################
#
# For details on how to install SDL/SDL2 see FAQ 3.8  - How do I compile an
# IOCCC entry that requires SDL1 or SDL2? which can be found at:
#
#   https://www.ioccc.org/faq.html#SDL
```

First of all the FAQ title is wrong. I can fix that. But right now that FAQ does not exist at the website linked to. Do you want me to put it as the temp-test-ioccc repo link for now? Or do you want me to do something else?

I'll fix it accordingly.

**Comment by lcn2** [comment]
> # Question about this wrt var.mk
> It has this:
> 
> ```makefile
> ##########################
> # SD1 and SDL2 libraries #
> ##########################
> #
> # For details on how to install SDL/SDL2 see FAQ 3.8  - How do I compile an
> # IOCCC entry that requires SDL1 or SDL2? which can be found at:
> #
> #   https://www.ioccc.org/faq.html#SDL
> ```
> 
> First of all the FAQ title is wrong. I can fix that. But right now that FAQ does not exist at the website linked to. Do you want me to put it as the temp-test-ioccc repo link for now? 

Yes, please.

**Comment by xexyl** [comment]
> > # Question about this wrt var.mk
> > It has this:
> > ```makefile
> > ##########################
> > # SD1 and SDL2 libraries #
> > ##########################
> > #
> > # For details on how to install SDL/SDL2 see FAQ 3.8  - How do I compile an
> > # IOCCC entry that requires SDL1 or SDL2? which can be found at:
> > #
> > #   https://www.ioccc.org/faq.html#SDL
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > First of all the FAQ title is wrong. I can fix that. But right now that FAQ does not exist at the website linked to. Do you want me to put it as the temp-test-ioccc repo link for now?
> 
> Yes, please.

Done.

**Comment by xexyl** [comment]
Should the newer entry json files FAQs be moved to the submit section? Probably there should but I guess it might be better to wait. We might even want another section entirely about the mkiocccentry toolkit. 

Which brings me to another comment below ..

**Comment by xexyl** [comment]
Should the FAQ have entries about the mkiocccentry toolkit tools?

I think it is a good idea and might help with the guidelines and maybe rules too.  

This can come later of course as the html issue is more important though this is related to it too. 

But I think mentioning txzchk and the others should be discussed in the FAQ. 

If you agree please advise on how you want it addressed and also when you think it SHOULD BE addressed.

**Comment by lcn2** [comment]
> Should the newer entry json files FAQs be moved to the submit section? Probably there should but I guess it might be better to wait. We might even want another section entirely about the mkiocccentry toolkit. 
> 
> Which brings me to another comment below ..

We do not recommend reorganizing the FAQ set until the links 🔗 to individual FAQs as they exist now, are changed to some standardization. 

Determine how links to FAQs are to be made, change everything to use the new standard. Then consider reorganizing the FAQ. 

**NOTE**: This issue has nothing to do with a FAQ reorganization directly. However one of the reasons why this issue exists is so that a reorganization of the FAQ can be done in the future (several times if needed 😝) without breaking lots of FAQ links and without having to change the "link text" of FAQ links elsewhere.

**Comment by lcn2** [comment]
> Should the FAQ have entries about the mkiocccentry toolkit tools?
> 
> I think it is a good idea and might help with the guidelines and maybe rules too.  

This issue has nil to do with the content of individual FAQs.  This issue is about **HOW TO LINK TO AN FAQ**. 

Discussion about adding a new FAQ and about improving an existing FAQ is probably better done under issue #2006.

**Comment by lcn2** [comment]
If there is interest in reorganizing the FAQ set, then let that interest drive the effort to **complete** this issue,.  **hint** 🤓

**Comment by xexyl** [comment]
> > Should the FAQ have entries about the mkiocccentry toolkit tools?
> > 
> > I think it is a good idea and might help with the guidelines and maybe rules too.  
> 
> This issue has nil to do with the content of individual FAQs.  This issue is about **HOW TO LINK TO AN FAQ**. 
> 
> Discussion about adding a new FAQ and about improving an existing FAQ is probably better done under issue #2006. 

I almost did put it there but since I was also bringing up the other part I thought that I would put it here. Okay so I can copy it over to there if that's helpful though that might or might not happen today.

**Comment by xexyl** [comment]
> > Should the newer entry json files FAQs be moved to the submit section? Probably there should but I guess it might be better to wait. We might even want another section entirely about the mkiocccentry toolkit. 
> 
> We do not recommend reorganizing the FAQ set until the links 🔗 to individual FAQs as they exist now, are changed to some standardization. 
> 
> Determine how links to FAQs are to be made, change everything to use the new standard. Then consider reorganizing the FAQ. 
> 
> **NOTE**: This issue has **nothing to do with a FAQ reorganization** directly. However one of the reasons why this issue exists is so that a reorganization of the FAQ can be done in the future (several times if needed 😝) without breaking lots of FAQ links and without having to change the "link text" of FAQ links elsewhere.

Understood. I will not worry about that part yet then.

**Comment by xexyl** [comment]
> If there is interest in reorganizing the FAQ set, then let that interest drive the effort to **complete** this issue,.  **hint** 🤓

Well now that you put it as high priority I will look at it more perhaps in between working on the other issue.

**Comment by xexyl** [comment]
It might actually prove that this could be completed more easily than the other since it it's not about making sure that the formatting and answers are good (that's part of the other issue and has to happen at the end). 

Since they are both high priority would you like me to (if I can) focus more on this issue so it can be closed as complete ?

**Comment by lcn2** [comment]
> It might actually prove that this could be completed more easily than the other since it it's not about making sure that the formatting and answers are good (that's part of the other issue and has to happen at the end). 
> 
> Since they are both high priority would you like me to (if I can) focus more on this issue so it can be closed as complete ?

We increased the priority of this issue because, once this issue is resolved, a FAQ organization improvement can be considered.

We will create a new "improve the organization of the FAQ" issue **once this issue is resolved**.

Until this issue is resolved, and while we are adding a few more things to the FAQ, such a "improve the organization of the FAQ" issue is premature.

**Comment by lcn2** [comment]
**FYI**: We have added clarification text to the top level TODO comment for this issue.

**Comment by xexyl** [comment]
> To resolve this issue
>
> Once the "how to link to an FAQ" has been determined, to close this issue we will have to go ahead the entire repo and change **all **links to FAQs accordingly.


Hmm .... I thought we already did decide on how to do it? Please clarify what you mean here as this point seems important.

**Comment by lcn2** [comment]
> > To resolve this issue
> 
> >
> 
> > Once the "how to link to an FAQ" has been determined, to close this issue we will have to go ahead the entire repo and change **all **links to FAQs accordingly.
> 
> 
> 
> 
> 
> Hmm .... I thought we already did decide on how to do it? Please clarify what you mean here as this point seems important.

If you believe there is a general design on how to refer to FAQs, then we suggest 2 things:

0. Restate the "how to link in a new comment"

.. so we can be sure we have made the right choice ... 

And once (0) has been done:

1. Modify some links to FAQS in a small number of non-FAQ pages

... so we can see the impact of that choice in a few pages before we start changing all the FAQ references.

**Comment by xexyl** [comment]
> > > To resolve this issue
> > 
> > 
> > > 
> > 
> > 
> > > Once the "how to link to an FAQ" has been determined, to close this issue we will have to go ahead the entire repo and change **all **links to FAQs accordingly.
> > 
> > 
> > Hmm .... I thought we already did decide on how to do it? Please clarify what you mean here as this point seems important.
> 
> If you believe there is a general design on how to refer to FAQs, then we suggest 2 things:
> 
> 0. Restate the "how to link in a new comment"
> 


I think the way already suggested for LINKING is a good idea (I think we both individually thought of it actually - I certainly had already started doing it).

> .. so we can be sure we have made the right choice ...
> 
> And once (0) has been done:
> 
> 1. Modify some links to FAQS in a small number of non-FAQ pages
> 

Do you mean the link text? If so I have already done this - and have done so when having to add new references to FAQs. I am pretty sure this is included in the guidelines if you want to look there. Possibly rules too but unsure.

> ... so we can see the impact of that choice in a few pages before we start changing all the FAQ references.

If I misunderstand the above, however, please clarify.

**Comment by lcn2** [comment]
> > > > To resolve this issue
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > Once the "how to link to an FAQ" has been determined, to close this issue we will have to go ahead the entire repo and change **all **links to FAQs accordingly.
> > > 
> > > 
> > > Hmm .... I thought we already did decide on how to do it? Please clarify what you mean here as this point seems important.
> > 
> > 
> > If you believe there is a general design on how to refer to FAQs, then we suggest 2 things:
> > 
> > 0. Restate the "how to link in a new comment"
> 
> I think the way already suggested for LINKING is a good idea (I think we both individually thought of it actually - I certainly had already started doing it).
> 
> > .. so we can be sure we have made the right choice ...
> > And once (0) has been done:
> > 
> > 1. Modify some links to FAQS in a small number of non-FAQ pages
> 
> Do you mean the link text? If so I have already done this - and have done so when having to add new references to FAQs. I am pretty sure this is included in the guidelines if you want to look there. Possibly rules too but unsure.
> 
> > ... so we can see the impact of that choice in a few pages before we start changing all the FAQ references.
> 
> If I misunderstand the above, however, please clarify.

We believe you misunderstand.  Please review the top comment.  Please look the current links to FAQs used throughout this repo.

**Comment by lcn2** [comment]
With commit d852734d982e60914ba2c15d9f9b60c6391afd28

fixed faq link URLs what used faq numbers

**Comment by lcn2** [comment]
# improve markdown references to FAQ entries

See commit 0bc6c3e4134d13fb9644b1493fce9762bff260eb

We quasi-canonicalized markdown links to FAQ entries.

**IMPORTANT NOTE**: This is a useful step in completing issue #2417.  This commit is consistent with the top level TODO comment requirements.  Nevertheless, this changes made by this commit needs to be examined and in some cases improved.

A link to an FAQ entry is of the form:

```
FAQ on "[the link text](....faq.html#string)"
    FAQ on "[the link text](....faq.html#string)"
```

The "FAQ" is at the start of the line, or in rare cases is preceded only by whitespace.

The reference to the FAQ starts outside of the markdown link with the string **FAQ ON "**.

After the markdown link is a **"** followed by either the end of line or a single punctuation character such as a period or comma.

We reduced the size of "the link text" that refer to an FAQ to at most 3 words.  In some cases the sentence and/or text surrounding the markdown link to a FAQ may need to be further adjusted to better accommodate this new style.

**IMPORTANT NOTE**: References to a FAQ entry are on a separate line.

p.s. When linking to the `faq.html` file itself, we use markdown of the form "`the [IOCCC FAQ](..../faq.html)`" and place the word "the" outside of the "IOCCC FAQ" link text.

## Useful command

Given the above system (and because references to an FAQ entry is on a separate line), the following command shows links to FAQ entries, sorted by FAQ entry "link #name" so that one may evaluate the way a given FAQ entry is referenced:

```sh
find * -name '*.md' -print0 | xargs0 grep -E '^[[:space:]]*FAQ on "' | sed -e 's/^\([^:]*\):\( *FAQ on "[^#]*\)#\([^)]*\)\()".*\)$/#\3\t\1\t\2#\3\4/' | sort
```

The above command shows the value of consistency when it comes to FAQ references.

**NOTE**: There are a few differences on how a given FAQ is references.  That is probably OK as there is NOT an "total mandate" that all references to a given FAQ must have the same "link text" for example.  On the other hand if references to a given FAQ can be identical, that might be helpful.  We are certain that in some contexts the "link text" might need to change.

Should we decide to change the way FAQ are referenced, we will be able to modify identify and modify them with greater ease.'

# TODO - What else is needed to close this issue

Examine the changes made by this commit and adjust and/or improve as needed.

Performed `make www` to accomplish the above and to test under macOS.

**Comment by xexyl** [comment]
BTW I think having in the link text `FAQ on` reads more naturally. That is why I have done it. The link is telling you what it is and the FAQ is part of it so it's good to have that in the link text. Anyway I guess we'll see how it looks. I'm sure it'll be fine and perhaps that's just a personal preference (but if so it'll take some effort to not include that part in the link text when doing more). But as the other issue is more important you say I'll worry about this later - though if I do come up with one I have to link to that might be something to consider.

Please advise on what you think with this so I know.

**Comment by lcn2** [comment]
> BTW I think having in the link text `FAQ on` reads more naturally. That is why I have done it. The link is telling you what it is and the FAQ is part of it so it's good to have that in the link text. Anyway I guess we'll see how it looks. I'm sure it'll be fine and perhaps that's just a personal preference (but if so it'll take some effort to not include that part in the link text when doing more). But as the other issue is more important you say I'll worry about this later - though if I do come up with one I have to link to that might be something to consider.
> 
> Please advise on what you think with this so I know.

The problem with making `FAQ on` part of the link text is that one would also need to include both double quotes, whereas keeping `FAQ on` as part of the enclosing sentence means that only what is inside the double quotes becomes the actual link.

**Comment by xexyl** [comment]
> > BTW I think having in the link text `FAQ on` reads more naturally. That is why I have done it. The link is telling you what it is and the FAQ is part of it so it's good to have that in the link text. Anyway I guess we'll see how it looks. I'm sure it'll be fine and perhaps that's just a personal preference (but if so it'll take some effort to not include that part in the link text when doing more). But as the other issue is more important you say I'll worry about this later - though if I do come up with one I have to link to that might be something to consider.
> > Please advise on what you think with this so I know.
> 
> The problem with making `FAQ on` part of the link text is that one would also need to include both double quotes, whereas keeping `FAQ on` as part of the enclosing sentence means that only what is inside the double quotes becomes the actual link.

Well in the changes I made I did not include it but I wonder why we need the quotes anyway. It seems not a good idea. With the link being a link it's clear what it is - as long as the link text is good.

**Comment by lcn2** [comment]
> Well in the changes I made I did not include it but I wonder why we need the quotes anyway. It seems not a good idea. With the link being a link it's clear what it is - as long as the link text is good.

Well by doing:

```
FAQ on "[some link text](.../faq.html#string)"
```

It is clear the "some link text" is the link and the double quotes contain the link text from the perspective to the containing sentence.  From the preceding _FAQ on_ it is clear that what is being referred within the double quotes **is** a FAQ.  And finally the displayed text is compact.

We believe this is an effective solution that also meets the top level TODO for this issue.

**Comment by xexyl** [comment]
> > Well in the changes I made I did not include it but I wonder why we need the quotes anyway. It seems not a good idea. With the link being a link it's clear what it is - as long as the link text is good.
> 
> Well by doing:
> 
> ```
> FAQ on "[some link text](.../faq.html#string)"
> ```
> 
> It is clear the "some link text" is the link and the double quotes contain the link text from the perspective to the containing sentence. From the preceding _FAQ on_ it is clear that what is being referred within the double quotes **is** a FAQ. And finally the displayed text is compact. For example:

I mean I can kind of see the first part but the link styling I think means that shouldn't be necessary. Well anyway this is not that important I suppose even if it looks somehow wrong to me.

> 
> ```
> Seem also the
> FAQ on "[some link text](.../faq.html#string)"
> for more information.
> ```
> 
> By keeping the _FAQ on "some link text"_ on a separate line, references to a FAQ ca be easily identified and modified if needed.
> 
> We believe this is an effective solution that also meets the top level TODO for this issue.

The second part does make sense .. I did not realise you had done that. I can't fix that now - have to go. I can fix what I did today, tomorrow, though.

Have a great day.

**Comment by lcn2** [comment]
With commit 89e66b96c5dd652f8c13bb789da28561b8a42ce7 improving the word flow around references to FAQs, we will close this issue.

Any additional minor FAQ reference refinements may be performed under issue #2006 in keeping with the [top comment](https://github.com/ioccc-src/temp-test-ioccc/issues/2417#issue-2347894671).
